File: project-euler.java/src/main/java/euler/solutions/_25.java
package euler.solutions;

import euler.common.BaseSolution;

import java.math.BigInteger;

public class _25 extends BaseSolution {

    @Override
    public long solve() {
        BigInteger a = new BigInteger("1"),
                   b = new BigInteger("1");
        long ord = 2;

        while (true) {
            BigInteger c = a.add(b);
            a = b;
            b = c;
            ord++;
            if (c.toString().length() == 1000) {
                break;
            }
        }

        return ord;
    }
}

File: project-euler.go/solutions/_25/25.go
package _25

import "math/big"

func Solve() int64 {
	a := big.NewInt(1)
	b := big.NewInt(1)
	ord := 2

	for {
		c := big.NewInt(0).Add(a, b)
		a = b
		b = c
		ord++
		if len(c.String()) == 1000 {
			break
		}
	}

	return int64(ord)
}
